RealmSwift/SwiftUI.swift

```diff
@@ -766,51 +766,149 @@ extension EnvironmentValues {
     }
 }
 
+/**
+An enum representing different states from `AsyncOpen` and `AutoOpen` process
+*/
+public enum AsyncOpenState {
+    /// Starting the Realm.asyncOpen process.
+    case connecting
+    /// Waiting for a user to be logged in before executing Realm.asyncOpen.
+    case waitingForUser
+    /// The Realm has been opened and is ready for use. For AsyncOpen this means that the Realm has been fully downloaded, but for AutoOpen the existing local file may have been used if the device is offline.
+    case open(Realm)
+    /// The Realm is currently being downloaded from the server.
+    case progress(Progress)
+    /// Opening the Realm failed.
+    case error(Error)
+}
+
+/**
+An enum representing the type of Async operation `@AsyncOpen` or `@AutoOpen`
+*/
+private enum AsyncOpenKind {
+    /// @AsyncOpen
```

Comment:
Since this is private it's fine to just leave it undocumented if there's nothing useful to say.